### PR TITLE
Final Capstone: BookaDeV

### PR DESCRIPTION
**Pull request by TahenyBELGUITH**
Hello 😄 
**In this project, we chose "renting developers " as a theme**

-  We set up the repos for the front-end and the back-end.
- We Installed the necessary dependencies for each side.
-  We created the API endpoints.
-  We created the front-end pages and made them responsive. We styled them using Boostrap and CSS
-  We made API requests and displayed content based on the response we got.
-  We created tests for both the front-end and the back-end.
 - We created the API's documentation using `swagger`.
Link to the API's PR 👉https://github.com/gak92/book_devs_api/pull/23
Thank you for the review 😊

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,230 +1,161 @@
...
+    <li><a href="https://rubyonrails.org/">React</a></li>
+    <li><a href="https://www.ruby-lang.org/en/">Redux</a></li>
   </ul>
 </details>
 
 <details>
   <summary>Server</summary>
   <ul>
-    <li><a href="https://expressjs.com/">Express.js</a></li>
+    <li><a href="https://netlify.com/">Netlify</a></li>
```
Comment (mcrd25) [inline README.md:59]: Maybe you can specify this is the deployment link

File: src/components/devs/addDeveloper.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+              className="form-control"
+              onChange={(e) => {
+                setRating(e.target.value);
+              }}
+            />
+            <label htmlFor="rating">rating</label>
+          </div>
+          {/* eslint_disable_anchor-is-valid */}
+          {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
+          <a href="#">
```
Comment (mcrd25) [inline src/components/devs/addDeveloper.js:137]: [OPTIONAL] Is there a cleaner way to write this?

File: src/components/sideNav.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+                    icon="lock"
+                    onClick={logoutUser}
+                    backgroundColor="#f44336"
+                  >
+                    Logout
+                  </CDBSidebarMenuItem>
+                </NavLink>
+              </>
+            ) : (
+              <>
```
Comment (mcrd25) [inline src/components/sideNav.js]: There ay be a better way to write this code to make it cleaner and neater. A looping method with arrays? try to see if you can make this cleaner [OPTIONAL]

**Review by mcrd25 — CHANGES_REQUESTED** [review]
Hi  Team, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
I've separated the changes requested below into front-end and back-end since I am not seeing a Back-end PR (Please add for next review)


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
### Front-end
- Don't forget that the application needs to look exactly like design with exception of photos, titles and descriptions. Regroup and maybe consider which components you can improve to look more like the design
- There are some candidates to make some of these html components reusable.
- Consider if there are areas you can follow REACT/JS best practices. Can use following articles as references:
    - https://www.freecodecamp.org/news/best-practices-for-react/
    - https://betterprogramming.pub/21-best-practices-for-a-clean-react-project-df788a682fb

### Back-end
- When I run the tests I get the three failures below: 
image




Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by gak92** [comment]
@mcrd25,
Thanks for the review

File: src/components/devs/addDeveloper.js (hunk 2 above)
Comment (Owusu-Desmond) [reply src/components/devs/addDeveloper.js:137]: Hi @mcrd25 

Can you some us a cleaner way of doing this by recommending any video or documentation for us

File: src/components/sideNav.js (hunk 3 above)
Comment (Owusu-Desmond) [reply src/components/sideNav.js]: Thank @mcrd25

**Review by mcrd25 — APPROVED** [review]
Hi Team, 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Comment by gak92** [comment]
@mcrd25,
Thanks for the review.

**Pull request merged**